src/Workspaces/Core/Portable/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.AnalysisData.cs

```diff
@@ -61,18 +54,18 @@ private abstract class AnalysisData : IDisposable
             ///        Value = 'true', because value assigned to 'x' here **may be** read on
             ///        some control flow path.
             /// </summary>
-            protected PooledDictionary<(ISymbol symbol, IOperation operation), bool> SymbolsWriteBuilder { get; }
+            protected abstract PooledDictionary<(ISymbol symbol, IOperation operation), bool> SymbolsWriteBuilder { get; }
```

Comment:
😕 It's not clear why we want to change these to `abstract`. It seems easier to leave them defined in the base class.

Reply:
I moved them to sub-type as the allocation and lifetime is controlled by sub-type. Especially for the sub-type FlowGraphAnalysisData, where we share these across multiple instances of FlowGraphAnalysisData. I don't have any real issue in keeping these defined in base type, while dispose is done in sub-type, if you feel that's more clear.

Reply:
I see. In that case I would agree `abstract` is better.